### ISSUE TITLE
Cannot auto-enable highlighting

### ISSUE THREAD

**Issue by antaed** [RRethy/vim-hexokinase#19]
Even though I have set `let g:Hexokinase_ftAutoload = ['*']` in my gvimrc, I have to run :HexokinaseToggle to turn on highlighting

I don't know if it's related or not but there's also this weird behaviour in **css files only**:

![hexokinase](https://user-images.githubusercontent.com/30736986/55254883-c14f0f00-5261-11e9-8bbe-f69da7e5af96.gif)

I tried to toggle the highlighting on/off, even to delete and reopen the buffer but I can only deactivate the background highlighting if I reopen gVim.

**Comment by RRethy** [comment]
1. Not sure why `let g:Hexokinase_ftAutoload = ['*']` is not working. I haven't been able to reproduce, what are all the hexokinase settings you have in your vimrc?
2. The video you displayed with background highlighting is not something that vim-hexokinase does so I'm not sure how it is doing that. Do you have a custom defined pattern? The sign column in your gif is definitely vim-hexokinase, but changing the background colour of the hex code is not part of this plugin unless you added that yourself.

**Comment by antaed** [comment]
1. This is in my gvimrc:
`let g:Hexokinase_highlighters = ['sign_column']`
`let g:Hexokinase_signIcon = '■'`
`let g:Hexokinase_ftAutoload = ['*']`
`let g:Hexokinase_refreshEvents = ['BufWritePost']`


2. I used another plugin (vim-css-color) before but I unistalled it as soon as I installed Hexakinase. I have no idea what's causing this to happen but I'll investigate further.

**Comment by RRethy** [comment]
1. Try changing `let g:Hexokinase_signIcon = '■'` to `let g:Hexokinase_signIcon = 'o'` and see if the new icon shows in the sign column. I don't think the setting are being applied since those settings for me work. Make sure you have it in the correct vimrc.
2. Yea the background highlighting is %100 vim-css-color.

**Comment by antaed** [comment]
I already tried to set a letter but still can't auto-load the highlighting.
Not a big issue though, I put this into my vimrc:
`autocmd FileType css silent! HexokinaseToggle`

**Comment by RRethy** [comment]
Did setting a letter work show a letter in the sign column? If it didn't, then your vim config is not being loaded and it isn't a hexokinase issue.

**Comment by antaed** [comment]
Can't auto-load highlighting with `let g:Hexokinase_ftAutoload` no matter what I do but it works fine with the autocmd from my previous post.

I just found the culprit for the weird behaviour in css files. The color code highlighting is caused by coc.nvim and I managed to turn it off by adding this to my coc-settings.json:
`"coc.preferences.colorSupport": false,`

**Comment by RRethy** [comment]
> Did setting a letter work show a letter in the sign column?

Did it? It's important since I don't think that configuration is being picked up by Vim. Add `let g:Hexokinase_signIcon = 'o'` **directly below or above** the Hexokinase_ftAutoload setting and see if the sign column shows that letter.

**Comment by antaed** [comment]
The letter does show up in the sign column but only after I run the HexokinaseToggle command, highlighting still won't auto-load. I even tried to put the Hexokinase settings right at the beginning/end of my vimrc.

**Comment by RRethy** [comment]
Interesting. It's a little hard guessing since I can't see your dotfiles. You can try adding `echom` statements to your local copy of the plugin, the relevant code is in `plugin/hexokinase.vim` lines 46-58. Might be useful to check `echom string(g:Hexokinase_ftAutoload)`. If that's too much work and ur fine with the autocmd you have then close the issue since I can't reproduce this with the provided info and this doesn't seem to be happening for others.

**Comment by antaed** [comment]
The `echom` statement returns an empty message no matter where I put it between lines 46-58 (except inside the `if !empty` loop, of course).
No problem though, I'm fine with the autocmd so I'm closing the issue. Thanks for your excellent work on this project and for your help.
